web3/process.go

```diff
@@ -119,91 +138,145 @@ func (c *Contracts) StateRoot(processID []byte) (*types.BigInt, error) {
 // submission, or an error if the submission fails. The tx hash can be used to
 // track the status of the transaction on the blockchain.
-func (c *Contracts) SetProcessTransition(processID, proof, inputs []byte, blobsSidecar *gtypes.BlobTxSidecar) (*common.Hash, error) {
+func (c *Contracts) SetProcessTransition(processID types.HexBytes, proof, inputs []byte, blobsSidecar *gtypes.BlobTxSidecar) (types.HexBytes, *common.Hash, error) {
+	// Copy processID into a fixed-size array to match the contract's expected
+	// type
 	var pid [32]byte
 	copy(pid[:], processID)
-	ctx, cancel := context.WithTimeout(context.Background(), web3QueryTimeout)
+	ctx, cancel := context.WithTimeout(context.Background(), web3WaitTimeout)
 	defer cancel()
-	if blobsSidecar == nil { // Regular transaction if no blobs are provided
-		autOpts, err := c.authTransactOpts()
-		if err != nil {
-			return nil, fmt.Errorf("failed to create transact options: %w", err)
-		}
-		autOpts.Context = ctx
-		tx, err := c.processes.SubmitStateTransition(autOpts, pid, proof, inputs)
-		if err != nil {
-			return nil, fmt.Errorf("failed to submit state transition: %w", err)
-		}
-		hash := tx.Hash()
-		return &hash, nil
-	}
-
+	// Prepare the ABI for packing the data
 	processABI, err := c.ProcessRegistryABI()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get process registry ABI: %w", err)
+		return nil, nil, fmt.Errorf("failed to get process registry ABI: %w", err)
 	}
-	tx, err := c.NewEIP4844Transaction(
-		ctx,
-		c.ContractsAddresses.ProcessRegistry,
-		processABI,
-		"submitStateTransition",
-		[]any{pid, proof, inputs},
-		blobsSidecar,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create EIP-4844 transaction: %w", err)
+
+	// Use transaction manager for automatic nonce management
+	var sentTx *gtypes.Transaction
+	txID, txHash, err := c.txManager.SendTx(ctx, func(nonce uint64) (*gtypes.Transaction, error) {
+		internalCtx, cancel := context.WithTimeout(context.Background(), web3WaitTimeout)
+		defer cancel()
+		// Build the transaction based on whether blobs are provided
+		switch blobsSidecar {
+		case nil: // Regular transaction
+			data, err := processABI.Pack("submitStateTransition", pid, proof, inputs)
+			if err != nil {
+				return nil, fmt.Errorf("failed to pack data: %w", err)
+			}
+			// No blobs so we dont not need to track sidecar, sentTx will be nil
+			return c.txManager.BuildDynamicFeeTx(internalCtx, c.ContractsAddresses.ProcessRegistry, data, nonce)
+		default: // Blob transaction
+			// Store tx in sentTx for tracking sidecar later
+			sentTx, err = c.NewEIP4844TransactionWithNonce(
+				internalCtx,
+				c.ContractsAddresses.ProcessRegistry,
+				processABI,
+				"submitStateTransition",
+				[]any{pid, proof, inputs},
+				blobsSidecar,
+				nonce,
+			)
+			return sentTx, err
+		}
+	})
+	// If blob transaction sent successfully, store sidecar for recovery
+	if err == nil && blobsSidecar != nil && sentTx != nil {
```

Comment:
Why not adding this code inside the switch/case too? It will be cleaner.

Reply:
No. Within the existing switch/case

Reply:
That is not posible, as i said the existing switch/case is inside a function that build the tx. This function is a parameter of SendTx function, is not in the same scope of the blob tracking. Only when that function ends is when the sidecar of the blob should be tracked. It is because the sent of the tx could not fail, but the the tx get stuck.